appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -155,60 +155,96 @@ export class CertificateManager {
+            throw error;
+        }
+
+        // If the secret has data, decode it
+        if (secretsObj && secretsObj.data) {
+            try {
+                certificate = {
                     caCert: Buffer.from(secretsObj.data['ca.cert'], 'base64').toString('utf-8'),
                     caKey: Buffer.from(secretsObj.data['ca.key'], 'base64').toString('utf-8'),
                     tlsCert: Buffer.from(secretsObj.data['tls.cert'], 'base64').toString('utf-8'),
                     tlsKey: Buffer.from(secretsObj.data['tls.key'], 'base64').toString('utf-8')
                 };
-
-                return certificate;
+            } catch (error) {
+                logger.error('Failed to decode secret data from base64!' + JSON.stringify(error), operationId, this.requestMetadata);
+                await logger.SendEvent("CertificateBase64DecodeFailure", operationId, null, clusterArmId, clusterArmRegion, true, error);
+
+                certificate = {
+                    caCert: null,
+                    caKey: null,
+                    tlsCert: null,
+                    tlsKey: null
+                };
+                logger.info('Returning empty certificate data due to decode failure.', operationId, this.requestMetadata);
             }
-        } catch (error) {
-            logger.error(JSON.stringify(error), operationId, this.requestMetadata);
-            throw error;
+
+            return certificate;
         }
+
+        return null;
     }
 
     public async GetMutatingWebhookCABundle(operationId: string, kubeConfig: k8s.KubeConfig): Promise<string> {
+        let caBundle: string = null;
+        let mutatingWebhookObject: k8s.V1MutatingWebhookConfiguration;
         try {
             const webhookApi: k8s.AdmissionregistrationV1Api = kubeConfig.makeApiClient(k8s.AdmissionregistrationV1Api);
-            const mutatingWebhookObject: k8s.V1MutatingWebhookConfiguration = await webhookApi.readMutatingWebhookConfiguration({ name: MutatingWebhookConfigurationName });
-            if (!mutatingWebhookObject 
-                || !mutatingWebhookObject.webhooks 
-                || mutatingWebhookObject.webhooks.length !== 1 || !mutatingWebhookObject.webhooks[0].clientConfig)
-            {
-                throw new Error("MutatingWebhookConfiguration not found or is malformed!");
-            }
-
-            return Buffer.from(mutatingWebhookObject.webhooks[0].clientConfig.caBundle, 'base64').toString('utf-8');
+            mutatingWebhookObject = await webhookApi.readMutatingWebhookConfiguration({ name: MutatingWebhookConfigurationName });
+            
         } catch (error) {
             logger.error(`Failed to get MutatingWebhookConfiguration! ${JSON.stringify(error)}`, operationId, this.requestMetadata);
+            await logger.SendEvent("MutatingWebhookConfigurationFetchFailure", operationId, null, null, null, true, error);
```

Comment:
Why the nulls? Pass it in if we need it, much nicer when there's context attached to each event. You can always find the entire trace by operationId, but if you don't know and you're looking by something else you'll miss random events, not good.